src/network_models/power_flow_evaluation.jl

```diff
@@ -106,36 +123,99 @@ function _make_pf_input_map!(
+            available_keys,
+            temp_component_map,
+            pf_data_opt_container_map,
+        )
+        pf_e_data.input_key_map[category] = pf_data_opt_container_map
+    end
+    # the function to map HVDC power transfers as bus injections is not applicable in this case:
+    pf_data isa PFS.PSSEExporter && return
+    _add_two_terminal_elements_map!(sys, pf_data, available_keys, pf_e_data.input_key_map)
+    return
+end
+
+function _add_category_to_map!(
+    precedence::Vector{DataType},
+    available_keys::Vector{Pair{OptimizationContainerKey, Any}},
+    temp_component_map::Union{
+        Dict{DataType, Dict{String, Int}},
+        Dict{DataType, Dict{Union{Int64, String}, String}},
+    },
+    pf_data_opt_container_map::Union{
+        Dict{OptimizationContainerKey, Dict{String, Int}},
+        Dict{OptimizationContainerKey, Dict{Union{Int64, String}, String}},
+    },
+)
+    added_injection_types = DataType[]
+    for entry_type in precedence
+        for (key, val) in available_keys
+            if get_entry_type(key) === entry_type
+                comp_type = get_component_type(key)
+                # Skip types that have already been handled by something of higher precedence
+                if comp_type in added_injection_types
+                    continue
+                end
+                push!(added_injection_types, comp_type)
+
+                name_bus_ix_map = valtype(temp_component_map)()
+                comp_names =
+                    if (key isa ParameterKey)
+                        get_component_names(get_attributes(val))
+                    else
+                        axes(val)[1]
                     end
-                    pf_data_opt_container_map[key] = name_bus_ix_map
+                for comp_name in comp_names
+                    name_bus_ix_map[comp_name] =
+                        temp_component_map[comp_type][comp_name]
                 end
+                pf_data_opt_container_map[key] = name_bus_ix_map
             end
         end
-        pf_e_data.input_key_map[category] = pf_data_opt_container_map
+    end
+end
+
+function _add_two_terminal_elements_map!(
+    sys::PSY.System,
+    pf_data::PFS.PowerFlowData,
+    available_keys::Vector{Pair{OptimizationContainerKey, Any}},
+    input_key_map::Dict{Symbol, Dict{OptimizationContainerKey, Dict{String, Int64}}},
+)
+    category = :active_power
+    category ∈ pf_input_keys(pf_data) || return
+
+    map_component_from = _make_temp_component_map(
+        pf_data,
+        sys,
+        PSY.TwoTerminalHVDC,
+        PSY.get_from ∘ PSY.get_arc,
+    )
+    isempty(map_component_from) && return
+
+    map_component_to = _make_temp_component_map(
+        pf_data,
+        sys,
+        PSY.TwoTerminalHVDC,
+        PSY.get_to ∘ PSY.get_arc,
+    )
+
+    # here just HVDC components, but could be extended to other two-terminal devices
+
+    entry_types = [FlowActivePowerFromToVariable, FlowActivePowerToFromVariable]
```

Comment:
Not sure about this design of using `PF_INPUT_KEY_PRECEDENCES` for all entry types except these. Could be convinced

Reply:
I couldn't come up with a way to add the capability of writing results for two-terminal elements using existing approach in code.

Reply:
Thoughts on defining two more categories besides `:active_power` and putting them there?